[3.0] IAS support in Remote Services and IAS migration guide (#1047)
cap-js/docs: java/cqn-services/remote-services.md

          url: https://url-of-the-second-cap-application
```

The plain service binding of XSUAA or IAS does not contain the URL of the remote API. Therefore it needs to be explicitly configured in the `options` section.
As the URL is typically not known at development time, it can be alternatively defined as an environment variable `CDS_REMOTE_SERVICES_<name>_OPTIONS_URL`.

:::tip

> Remote APIs which require IAS-based authentication might expect a certificate based client authentication in addition to the IAS-based JWT token (ie. [proof-of-possession](https://github.com/SAP/cloud-security-services-integration-library/tree/629aef6f54cf0b9e41453eb32082a889c16b22d3/java-security#proofofpossession-validation)). 

Reviewer: ```suggestion
Remote APIs which require IAS-based authentication might expect a certificate based client authentication in addition to the IAS-based JWT token (ie. [proof-of-possession](https://github.com/SAP/cloud-security-services-integration-library/tree/main/java-security#proofofpossession-validation)). 
```

        name: s4-business-partner-api
```

If your CAP application is using IAS and you want to call a _remote API_ that is provided by another IAS-based application (ie. Application2Application scenario), you can utilize a simplified security configuration in the destination.
As a pre-requisite, your CAP application and the called application need to trust the same IAS tenant and you need to define a dependency in IAS to consume the respective API provided by the _remote API_.

Create a destination configuration with the following parameters:

> ```yaml

Reviewer 2: some part of this block needs to be made into a code block to work. I'm checking.

> ```yaml

Reviewer 2: a part of this needs to be in a code block, I'm checking.

Reviewer: Most likely the <> characters make problems..

Reviewer 2: yep, escaping with a backslash or using backticks fixes it. :)

> url: <url-of-the-remote-api>

Reviewer: Maybe better structure this as a bullet point list?

>     url: <url-of-the-remote-api>
>     authentication-type: None
>     additional property:
>         cloudsdk.ias-dependency-name: <name-of-the-ias-dependency>

Reviewer: ```suggestion
url: <url-of-the-remote-api>
authentication-type: None
additional property:
  cloudsdk.ias-dependency-name: <name-of-the-ias-dependency>
```

One might think that this is a configuration file you need to place somewhere. Maybe rather use a bullet point list?

If your CAP application is using IAS and you want to call a _remote API_ that is provided by another IAS-based application (ie. Application2Application scenario), you can utilize a simplified security configuration in the destination.
As a pre-requisite, your CAP application and the called application need to trust the same IAS tenant and you need to define a dependency in IAS to consume the respective API provided by the _remote API_.

Create a destination configuration with the following parameters:

> - `url`: <url-of-the-remote-api>
> - `authentication-type`: None
> - additional property:
>   - `cloudsdk.ias-dependency-name`: <name-of-the-ias-dependency>

Reviewer: Let's use the same terminology as in the Destinations UI of BTP Cockpit

```suggestion
- URL: <url-of-the-remote-api>
- Authentication: NoAuthentication
- Additional Properties:
  - `cloudsdk.ias-dependency-name`: <name-of-the-ias-dependency>
```
